### PR TITLE
Add link to Copilot in CLI printDiagnostics

### PR DESCRIPTION
**Pull request by arturl**
Add a deep link to Pulumi Console from CLI.
Details: https://docs.google.com/document/d/1QJucNEnO4hcnhqwADyMYXEHMuyhrB6tyulpoxbYX7Xg/edit?usp=sharing

### CODE REVIEW

File: pkg/backend/display/progress.go [hunk 1]
```diff
@@ -619,8 +622,17 @@ func (display *ProgressDisplay) printDiagnostics() bool {
 				display.println("")
 			}
 		}
+	}
 
+	// Print a link to Copilot to explain the failure.
+	// Check for SuppressPermalink ensures we don't print the link for DIY backends
+	if wroteDiagnosticHeader && !display.opts.SuppressPermalink && !display.opts.SuppressLinkToCopilot {
+		display.println("    " +
+			colors.SpecCreateReplacement + "[Pulumi Copilot]" + colors.Reset + " Would you like help with these diagnostics?")
```
Comment (AaronFriel) [inline pkg/backend/display/progress.go:631]: Diagnostics are emitted for a fairly wide variety of reasons - can we more precisely catch when an update fails instead?

File: pkg/backend/display/progress.go [hunk 2]
```diff
@@ -619,8 +622,17 @@ func (display *ProgressDisplay) printDiagnostics() bool {
...
+		display.println("    " +
+			colors.Underline + colors.Blue + display.permalink + "?explainFailure" + colors.Reset)
```
Comment (AaronFriel) [inline pkg/backend/display/progress.go:633]: It's not obvious to me why this is better than `?askCopilot`. Even if the query parameter was only a key, no value, I think that's clearer to me than `?explainFailure`.

Comment (arturl) [reply pkg/backend/display/progress.go:633]: > It's not obvious to me why this is better than `?askCopilot`. Even if the query parameter was only a key, no value, I think that's clearer to me than `?explainFailure`.

This is based on feedback from the team - url-encoded textual query looks 'ugly' in the CLI and requires a new release of the CLI (plus user upgrading their CLI) when we have to change it.

File: pkg/backend/display/progress.go (hunk 1 above)
Comment (arturl) [reply pkg/backend/display/progress.go:631]: This should work for any failure, the Copilot should be able to explain it based on the context in the permalink.

**Review by tgummerer — APPROVED** [review]

File: pkg/backend/display/progress.go [hunk 3]
```diff
@@ -619,8 +622,17 @@ func (display *ProgressDisplay) printDiagnostics() bool {
 				display.println("")
 			}
 		}
+	}
 
+	// Print a link to Copilot to explain the failure.
+	// Check for SuppressPermalink ensures we don't print the link for DIY backends
+	if wroteDiagnosticHeader && !display.opts.SuppressPermalink && !display.opts.SuppressLinkToCopilot {
```
Comment (julienp) [inline pkg/backend/display/progress.go:629]: How do we ensure we don't print this on non-cloud backends?

Comment (julienp) [reply pkg/backend/display/progress.go:629]: Oh I guess we have `display.opts.SuppressPermalink` in that case? Nice!

Comment (julienp) [reply pkg/backend/display/progress.go:629]: `SuppressCloudLinks` sounds good to me! The comment in the `Options` struct could be clearer too.

This PR is good to go as is though!

Comment (arturl) [reply pkg/backend/display/progress.go:629]: `SuppressPermalink` maps to the `--suppress-permalink` flag in for `pulumi up` (you will see it if you do `pulumi help up`), and changing that would be a breaking change (and requiring a doc update). Renaming `SuppressPermalink` without renaming `--suppress-permalink` would be unhelpful.

I will update the comment.

Thanks for the review!

File: pkg/backend/display/options.go [hunk 4]
```diff
@@ -57,6 +57,7 @@ type Options struct {
 	Stderr                 io.Writer           // the writer to use for stderr. Defaults to os.Stderr if unset.
 	SuppressTimings        bool                // true to suppress displaying timings of resource actions
 	SuppressProgress       bool                // true to suppress displaying progress spinner.
+	SuppressLinkToCopilot  bool                // true to suppress displaying a 'explainFailure' link to Copilot.
```
Comment (AaronFriel) [inline pkg/backend/display/options.go:60]: Should this be `ShowLinkToCopilot` to default it to false?

And should it then only be conditionally set to true when the user is using the Pulumi Cloud (api.pulumi.com) backend?

File: pkg/backend/display/progress.go (hunk 2 above)
Comment (AaronFriel) [reply pkg/backend/display/progress.go:633]: I'm suggesting we have it be `?askCopilot` with no URl encoded query. Just `?askCopilot`.

File: pkg/backend/display/options.go (hunk 4 above)
Comment (arturl) [reply pkg/backend/display/options.go:60]: `SuppressPermalink` will be set to `false` for DIY backend, this is the flag that's checked. 
We decided that the feature should be on by default, hence 'Suppress...' and not 'Show...' in the name of the flag.

**Pull request merged**